app/lib/cashu/use-animated-qr-encoder.tsx

```diff
@@ -0,0 +1,50 @@
+import type { Buffer } from 'node:buffer';
+import type { UREncoder } from '@gandlaf21/bc-ur';
+import { useEffect, useState } from 'react';
+import { useInterval } from 'usehooks-ts';
+
+type Props = {
+  text: string;
+  intervalMs?: number;
+};
+
+export function useAnimatedQREncoder({ text, intervalMs = 200 }: Props) {
+  const [fragment, setFragment] = useState<string>('');
+  const [encoder, setEncoder] = useState<UREncoder | null>(null);
+
+  const maxFragmentLength = 150;
+
+  useEffect(() => {
+    // need to import here because the UREncoder requires a cbor library that only works in the browser
+    import('@gandlaf21/bc-ur').then((bcUr) => {
+      const { UR, UREncoder } = bcUr;
+
+      const messageBuffer = new Uint8Array(
+        text.split('').map((c) => c.charCodeAt(0)),
+      );
+      const ur = UR.fromBuffer(messageBuffer as Buffer);
```

Comment:
I'm not sure how I was able to commit this before, but now I'm getting a type error that messageBuffer doesn't sufficiently overlap with Buffer. What do you think I should do here with the types?
Options I see:
1. bun install buffer and use that instead of a Uint8Array
2. remove the type casting `as Buffer` and ts-ignore it